### PR TITLE
Add PO Localization Sample for .NET 5

### LINKED ISSUE #28373: Update to .NET 6 or higher
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 6b1098b7-dbe7-a8d8-f57f-017b1c94e960
* Version Independent ID: 96179de2-df1c-ff1d-5a2e-89b0949e0620
* Content: [Configure portable object localization in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/portable-object-localization?view=aspnetcore-7.0)
* Content Source: [aspnetcore/fundamentals/portable-object-localization.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/portable-object-localization.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @sebastienros
* Microsoft Alias: **scaddie**

---
[Associated WorkItem - 63765](https://dev.azure.com/msft-skilling/Content/_workitems/edit/63765)

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #28373

### CODE REVIEW

**Comment by hishamco** [comment]
@Rick-Anderson could you please let me know why the build failed?

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson could you please let me know why the build failed?

I'll fix it tomorrow.

File: aspnetcore/fundamentals/portable-object-localization.md [hunk 1]
```diff
@@ -8,6 +8,424 @@ uid: fundamentals/portable-object-localization
...
+
+### Using a different default pluralized language
+
+The package includes a `Plural` extension method that's specific to two plural forms. For languages requiring more plural forms, create an extension method. With an extension method, you won't need to provide any localization file for the default language &mdash; the original strings are already available directly in the code.
+
+You can use the more generic `Plural(int count, string[] pluralForms, params object[] arguments)` overload which accepts a string array of translations.
+
+:::moniker-end
+
+:::moniker range=">= aspnetcore-5.0"
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/portable-object-localization.md:218]:
```suggestion
:::moniker range=">= aspnetcore-5.0 < aspnetcore-6.0"
```

File: aspnetcore/fundamentals/portable-object-localization.md [hunk 2]
```diff
@@ -8,6 +8,424 @@ uid: fundamentals/portable-object-localization
...
+
+### Using a different default pluralized language
+
+The package includes a `Plural` extension method that's specific to two plural forms. For languages requiring more plural forms, create an extension method. With an extension method, you won't need to provide any localization file for the default language &mdash; the original strings are already available directly in the code.
+
+You can use the more generic `Plural(int count, string[] pluralForms, params object[] arguments)` overload which accepts a string array of translations.
+
+:::moniker-end
+
+:::moniker range=">= aspnetcore-5.0 < aspnetcore-6.0"
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/portable-object-localization.md:218]: You'll need to `git pull` now

File: aspnetcore/fundamentals/portable-object-localization.md [hunk 3]
```diff
@@ -8,6 +8,424 @@ uid: fundamentals/portable-object-localization
...
+
+### Using a different default pluralized language
+
+The package includes a `Plural` extension method that's specific to two plural forms. For languages requiring more plural forms, create an extension method. With an extension method, you won't need to provide any localization file for the default language &mdash; the original strings are already available directly in the code.
+
+You can use the more generic `Plural(int count, string[] pluralForms, params object[] arguments)` overload which accepts a string array of translations.
+
+:::moniker-end
+
 :::moniker range=">= aspnetcore-3.0"
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/portable-object-localization.md:429]:
```suggestion
:::moniker range="< aspnetcore-5.0"
```

File: aspnetcore/fundamentals/portable-object-localization.md [hunk 4]
```diff
@@ -8,7 +8,425 @@ uid: fundamentals/portable-object-localization
 ---
 # Configure portable object localization in ASP.NET Core
 
-:::moniker range=">= aspnetcore-3.0"
+:::moniker range=">= aspnetcore-6.0"
+
+By [Sébastien Ros](https://github.com/sebastienros), [Scott Addie](https://twitter.com/Scott_Addie) and [Hisham Bin Ateya](https://github.com/hishamco)
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/portable-object-localization.md:13]:
```suggestion
By [Sébastien Ros](https://github.com/sebastienros) and [Hisham Bin Ateya](https://github.com/hishamco)
```

File: aspnetcore/fundamentals/portable-object-localization.md [hunk 5]
```diff
@@ -8,7 +8,425 @@ uid: fundamentals/portable-object-localization
 ---
 # Configure portable object localization in ASP.NET Core
 
-:::moniker range=">= aspnetcore-3.0"
+:::moniker range=">= aspnetcore-6.0"
+
+By [Sébastien Ros](https://github.com/sebastienros) and [Hisham Bin Ateya](https://github.com/hishamco)
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/portable-object-localization.md:13]:
```suggestion
By  [Hisham Bin Ateya](https://github.com/hishamco) and [Sébastien Ros](https://github.com/sebastienros).
```

**Comment by sebastienros** [comment]
Why add a sample for .NET 5 which is not supported?

**Comment by hishamco** [comment]
Seb I just added a missing sample for both .NET 5 & 6, I know it's not supported now, but it's useful for those who already have a running apps with older version

**Comment by Rick-Anderson** [comment]
> Why add a sample for .NET 5 which is not supported?

I agree. .NET 6 should be fine.

**Comment by Rick-Anderson** [comment]
> Seb I just added a missing sample for both .NET 5 & 6, I know it's not supported now, but it's useful for those who already have a running apps with older version

IF if's there, might as well keep it.

**Comment by hishamco** [comment]
We already keep track samples for all the versions

File: aspnetcore/fundamentals/portable-object-localization.md [hunk 6]
```diff
@@ -59,45 +265,45 @@ The PO file specification can be found [here](https://www.gnu.org/savannah-check
...
 
 Add the required middleware to the `Configure` method of `Startup.cs`:
 
 [!code-csharp[](localization/sample/3.x/POLocalization/Startup.cs?name=snippet_Configure&highlight=15)]
 
-Add the following code to your Razor view of choice. `About.cshtml` is used in this example.
+Add the following code to your Razor page of choice. `Index.cshtml` is used in this example.
 
-[!code-cshtml[](localization/sample/3.x/POLocalization/Views/Home/About.cshtml)]
+[!code-cshtml[](localization/sample/3.x/POLocalization/Pages/Index.cshtml)]
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/portable-object-localization.md:290]:
```suggestion
<!-- FIX this invalid file reference
[!code-cshtml[](localization/sample/3.x/POLocalization/Pages/Index.cshtml)] -->
```

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/portable-object-localization.md:290]: This is the last problem

**Comment by Rick-Anderson** [comment]
Moved to #28376, carried over all the commits so @hishamco  gets credit when it merges.

**Pull request closed without merging**